### PR TITLE
feat(destinations): validate no update on slack

### PR DESCRIPTION
**Pull request by nzur-newrelic**
# Description
The current documentation shows to use `ignore_changes` on slack destinations, but sadly this situation causes issues with our mutations, therefore update for it will never work.
To avoid this issue we'll allow users only to destroy the resource if needed.

A validation has been added for slack to avoid updating it as well in case someone didn't see the docs.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

Please delete options that are not relevant.

- [ ] My commit message follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] My code is formatted to [Go standards](https://go.dev/blog/gofmt)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes. Go [here](https://github.com/newrelic/terraform-provider-newrelic/blob/main/CONTRIBUTING.md#testing) for instructions on running tests locally.

## How to test this change?

Please describe how to test your changes. Include any relevant steps in the UI, HCL file(s), commands, etc

- Create a slack destination (through the UI)
- Import it to terraform
- Try to update the destination

### CODE REVIEW

**Review by asaff-signifai — APPROVED** [review]

File: website/docs/r/notification_destination.html.markdown [hunk 1]
```diff
@@ -243,51 +243,15 @@ resource "newrelic_notification_destination" "foo" {
 
 In order to create a Slack destination, you have to grant our application access to your workspace. This process is [based on OAuth](https://api.slack.com/authentication/oauth-v2) and can only be done through a browser.
 As a result, you cannot set up a Slack destination purely with Terraform code.
-You can either create a Slack destination in the interface and [import](#import) it to Terraform, or use destination data source.
+You can either use [the destination data source](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/data-sources/notification_destination),
+or reference destination id directly. Destination ID which can be found through [the destination data source](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/data-sources/notification_destination),
+our GraphQL API, or using the user interface.
 
+~> **NOTE:** Please be aware that while you can import the Slack destination resource, it can only be used to delete it through Terraform.
```
Comment (pranav-new-relic) [inline website/docs/r/notification_destination.html.markdown:250]: as discussed on Slack, this note can be shifted downwards, with a **warning** that states
 
```
Slack-based destinations configured/set up using the `newrelic_notification_destination` resource can only be imported and destroyed; this resource **does not** support creating/updating Slack-based destinations, owing to the reasons stated above, under the **Slack** section. Hence, **it is recommended to use the** `newrelic_notification_destination` data source instead, **to fetch the ID of the Slack destination configured via the UI** to use with other resources in the New Relic Terraform Provider, such as `newrelic_workflow`.
```

File: website/docs/r/notification_destination.html.markdown [hunk 2]
```diff
@@ -243,51 +243,15 @@ resource "newrelic_notification_destination" "foo" {
...
-  }
-  
-  property {
-      key   = "teamName"
-      label = "Team Name"
-      value = "******"
-  }
-}
-```
-
```
Comment (pranav-new-relic) [inline website/docs/r/notification_destination.html.markdown:290]: please revert, as discussed on Slack

File: newrelic/resource_newrelic_notifications_destination.go [hunk 3]
```diff
@@ -350,6 +350,12 @@ func resourceNewRelicNotificationDestinationUpdate(ctx context.Context, d *schem
 		return diag.FromErr(err)
 	}
 
+	destinationType := notifications.AiNotificationsDestinationType(d.Get("type").(string))
+
+	if isOAuth2SlackType(destinationType) {
+		return diag.FromErr(fmt.Errorf("a destination with '%s' type cannot be updated via terraform", destinationType))
+	}
```
[the remark this answers is not in the document]
Comment (pranav-new-relic) [reply newrelic/resource_newrelic_notifications_destination.go:357]: Sorry, I just realized the `type`of the destination has a `ForceNew` on it, else my suggestion was meant to create a (say) email based destination, and try updating it to Slack which would run into the new error you've added, but this isn't possible. Thanks for clarifying 👍

File: website/docs/r/notification_destination.html.markdown [hunk 4]
```diff
@@ -243,39 +243,40 @@ resource "newrelic_notification_destination" "foo" {
...
-You can either create a Slack destination in the interface and [import](#import) it to Terraform, or use destination data source.
-
+You can either use [the destination data source](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/data-sources/notification_destination),
+or reference destination id directly. Destination ID which can be found through [the destination data source](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/data-sources/notification_destination),
+our GraphQL API, or using the user interface.
 
 ## Import
 
+~> **WARNING:** Please be aware that while you can import the Slack destination resource, it can only be used to delete it through Terraform.
+
```
Comment (pranav-new-relic) [inline website/docs/r/notification_destination.html.markdown:253]: This is nice, but would you want to consider keeping this a little more elaborate, in order to help convey that it isn't really useful to use this destination via Terraform (along the lines of what I've posted [here](https://github.com/newrelic/terraform-provider-newrelic/pull/2661#discussion_r1607150178))? Just give this a thought :)

File: website/docs/r/notification_destination.html.markdown [hunk 5]
```diff
@@ -243,39 +243,40 @@ resource "newrelic_notification_destination" "foo" {
 
 In order to create a Slack destination, you have to grant our application access to your workspace. This process is [based on OAuth](https://api.slack.com/authentication/oauth-v2) and can only be done through a browser.
 As a result, you cannot set up a Slack destination purely with Terraform code.
-You can either create a Slack destination in the interface and [import](#import) it to Terraform, or use destination data source.
-
+You can either use [the destination data source](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/data-sources/notification_destination),
+or reference destination id directly. Destination ID which can be found through [the destination data source](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/data-sources/notification_destination),
+our GraphQL API, or using the user interface.
```
Comment (pranav-new-relic) [inline website/docs/r/notification_destination.html.markdown:248]:
```suggestion
However, if you would like to use Slack-based destinations with other resources in the New Relic Terraform Provider, the [data source `newrelic_notification_destination`](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/data-sources/notification_destination),
may be used to fetch the ID of the destination; alternatively, you might want to source the ID of the destination from  NerdGraph, or from the New Relic One UI.
```

(just trying to make this paragraph look better - please let me know if it sounds okay, too :))

Comment (nzur-newrelic) [reply website/docs/r/notification_destination.html.markdown:248]: There's no real need to use the slack-based destination on TF, you can just use the id itself, which is already needed to get this resource or data source in the first place.
So this update might make it a bit confusing...

File: website/docs/r/notification_destination.html.markdown (hunk 4 above)
Comment (nzur-newrelic) [reply website/docs/r/notification_destination.html.markdown:253]: I changed the wording a bit given that the warning is inside the import section so there's no need to talk about configuring or creating a destination. 
Regarding the data source, we already talk about it on the slack section so I removed it as I wanted to make the warning as clear as possible.

File: website/docs/r/notification_destination.html.markdown (hunk 5 above)
Comment (pranav-new-relic) [reply website/docs/r/notification_destination.html.markdown:248]: You're right, but I just wanted to rephrase your sentence, just to make it slightly more better grammatically (and not add anything else to the sentence); so I think we might want to have this change (maybe? Let me know what you think :)

File: website/docs/r/notification_destination.html.markdown [hunk 6]
```diff
@@ -243,39 +243,40 @@ resource "newrelic_notification_destination" "foo" {
...
 
 ## Import
 
+~> **WARNING:** Slack-based destinations can only be imported and destroyed; this resource **does not** support updating Slack-based destinations, owing to the reasons stated above, under the **Slack** section.
```
Comment (pranav-new-relic) [inline website/docs/r/notification_destination.html.markdown:252]:
```suggestion
~> **WARNING:** Slack-based destinations can only be imported and destroyed; this resource **does not** support creating and updating Slack-based destinations, owing to the reasons stated above, under the **Slack** section.
```

(one final nitpick)

File: website/docs/r/notification_destination.html.markdown (hunk 4 above)
Comment (pranav-new-relic) [reply website/docs/r/notification_destination.html.markdown]: https://github.com/newrelic/terraform-provider-newrelic/pull/2661#discussion_r1608398693

**Pull request merged**